src/pages/index.js

```diff
@@ -0,0 +1,264 @@
+    })
+    .catch((error) => {
+      console.error("Error in addNewCard:", error);
+      return Promise.reject(error);
+    })
+    .finally(() => {
+      newCardPopup.setLoading(false);
+    });
+}
+
+function handleProfileFormSubmit(userData) {
+  newEditPopup.setLoading(true);
+  api
+    .updateUserInfo(userData.name, userData.descripton)
+    .then(() => {
+      userInfo.setUserInfo(userData.name, userData.descripton);
+      newEditPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error in updateUserInfo:", error);
+      if (error instanceof SyntaxError) {
+        console.error("Response body:", error.body);
+      }
+    })
+    .finally(() => {
+      newEditPopup.setLoading(false);
+    });
+}
+
+/*
+handleDeleteButtonClick function will be called only after the click of the delete
+icon on the card that we want to delete
+the method will open the popup of the confirm modal and will wait for the submit 
+event , if the submit event is happaning only then the confirmationPopup.setSubmitAction()
+*/
+function handleDeleteButtonClick(card) {
+  confirmationPopup.open(); //opening the popup
+  confirmationPopup.setSubmitAction(() => {
+    confirmationPopup.setDeleteLoading(true); //cahnge the text content of the confirm button
+    api
+      .deleteCard(card._id)
+      .then((result) => {
+        card.removeCardElement(result); //card.js method to remove the card_element
+        if (result) {
+          confirmationPopup.close();
+        }
+      })
+      .catch((err) => {
+        console.error(`${err} Failed to delete post.`);
+        console.error("Error in deleteCard:", error);
+        return Promise.reject(error);
+      })
+      .finally(() => {
+        confirmationPopup.setDeleteLoading(false);
+      });
+  });
+}
+/*
+handleLikedButtonClick funtion will be called after the event click on the like button
+this function is passed to the card class
+at the begining its checking if the isLiked is true or false and by the state
+of the isLiked its decide's wich api method to call 
+after the api call we manualy changing the isLiked atribute of the img with the
+setLike function from the card class that adding or removing the like button
+active class
+*/
+function handleLikedButtonClick(card) {
+  if (card.isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then((res) => {
+        card.setLike(res.isLiked);
+      })
+      .catch((error) => {
+        console.error("Error in dislikeCard:", error);
+        return Promise.reject(error);
+      });
+  } else {
+    api
+      .likeCard(card._id)
+      .then((res) => {
+        card.setLike(res.isLiked);
+      })
+      .catch((error) => {
+        console.error("Error in the LikeCard:", error);
+        return Promise.reject(error);
+      });
+  }
+}
+
+function upDateAvatar(avatar) {
+  avatarPopup.setLoading(true);
+  api
+    .upDateAvater(avatar.link)
+    .then((res) => {
+      avatarImage.src = res.avatar;
```

Comment:
Class `UserInfo` should set the entire user’s data (including the **avatar**)

Reply:
fixed it now the userInfo updating everything